### PR TITLE
[terra-tabs] Update uncontrolled responsive-element to controlled responsive-element

### LINKED ISSUE #969: [terra-tabs] Uplift to the controlled responsive element
# Feature Request

Investigate uplifting terra-tabs onto the controlled responsive element or alternative. 

## Description

The uncontrolled responsive element has been [flagged for deprecation](https://github.com/cerner/terra-core/pull/2729). The terra-tabs should either be uplifting onto the controlled responsive element implementation or alternatively the responsive element should be removed from the component.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
resolves #969

### CODE REVIEW

File: packages/terra-tabs/src/Tabs.jsx [hunk 1]
```diff
@@ -187,16 +188,22 @@ class Tabs extends React.Component {
...
       <ContentContainer
         {...customProps}
         className={tabsClassNames}
         fill={fill}
         header={(
-          <ResponsiveElement
-            tiny={collapsedTabs}
-            small={collapsibleTabs}
-          />
+          <ResponsiveElement onChange={value => (this.setState({ breakpoint: value }))}>
```
Comment (emilyrohrbough) [inline packages/terra-tabs/src/Tabs.jsx]: Should we do something like the following so we only update state when transitioning between tiny and non-tiny?
```js
<ResponsiveElement
    onChange={value => {
        const showCollapsedTabs = value === 'tiny';

        if (this.state.showCollapsedTabs !== showCollapsedTabs) {
            this.setState({ showCollapsedTabs })
        }
    }}
>
```

Comment (pranav300) [reply packages/terra-tabs/src/Tabs.jsx]: updated here https://github.com/cerner/terra-framework/pull/972/commits/b6df56caadf61ab4705b2bb550fb93eb061642b8.

File: packages/terra-tabs/src/Tabs.jsx [hunk 2]
```diff
@@ -84,6 +84,7 @@ class Tabs extends React.Component {
     this.state = {
       activeKey: this.getInitialState(),
       isLabelTruncated: false,
+      showCollapsedTabs: undefined,
```
Comment (emilyrohrbough) [inline packages/terra-tabs/src/Tabs.jsx]: Should this be false?

Comment (pranav300) [reply packages/terra-tabs/src/Tabs.jsx]: Updated here https://github.com/cerner/terra-framework/pull/972/commits/e2424685bd0d35297cd4932e47e57f1b0032bedd.

File: packages/terra-tabs/tests/jest/Tabs.test.jsx [hunk 3]
```diff
@@ -7,62 +7,79 @@ describe('Tabs', () => {
...
     const wrapper = shallowWithIntl(defaultRender);
+    wrapper.update();
     expect(wrapper).toMatchSnapshot();
   });
 
   describe('Variants', () => {
     it('should render modular-left-aligned tabs', () => {
       const defaultRender = <Tabs variant="modular-left-aligned"><Tabs.Pane label="Default" key="default" /></Tabs>;
       const wrapper = shallowWithIntl(defaultRender);
+      wrapper.setState({ showCollapsedTabs: true });
```
Comment (nramamurth) [inline packages/terra-tabs/tests/jest/Tabs.test.jsx]: I believe that these "Variants" tests could be removed - there is no `variant` prop.

Comment (emilyrohrbough) [reply packages/terra-tabs/tests/jest/Tabs.test.jsx]: @nramamurth should we log an issue to clean it up?

Comment (nramamurth) [reply packages/terra-tabs/tests/jest/Tabs.test.jsx]: @emilyrohrbough I remember that there was a discussion to remove modular tabs implementation on one of the PRs and Ryan had mentioned that we could do that as a separate issue. I am not sure if we have an issue logged to do that though. I shall go ahead and create one and document the this detail as well if there ain’t one.

**Comment by emilyrohrbough** [comment]
Changed to tech-only since we've treated the other responsive-element PRs as tech only.

**Pull request merged**